transport/src/test/java/io/netty/channel/nio/NioEventLoopTest.java

```diff
@@ -132,4 +135,37 @@ public void run() {
             group.shutdownGracefully();
         }
     }
+
+    @Test(timeout = 1000)
+    public void testArbitrarySelectableChannel() throws Exception {
+        NioEventLoopGroup group = new NioEventLoopGroup(1);
+        NioEventLoop loop = (NioEventLoop) group.next();
+
+        try {
+            Channel channel = new NioServerSocketChannel();
+            loop.register(channel).syncUninterruptibly();
+            channel.bind(new InetSocketAddress(0)).syncUninterruptibly();
+
+            SocketChannel selectableChannel = SocketChannel.open();
+            selectableChannel.configureBlocking(false);
+            selectableChannel.connect(channel.localAddress());
+
+            final CountDownLatch latch = new CountDownLatch(1);
+
+            loop.register(selectableChannel, SelectionKey.OP_CONNECT, new NioTask<SocketChannel>() {
+                @Override
+                public void channelReady(SocketChannel ch, SelectionKey key) {
+                    latch.countDown();
+                }
+
+                @Override
+                public void channelUnregistered(SocketChannel ch, Throwable cause) {
+                }
+            });
+
+            latch.await();
```

Comment:
Await with timeout?

Reply:
That could be done, but please note that I set a timeout on the `@Test` annotation so that the whole test case will fail in case of a timeout.

Reply:
Works for me.